arangod/MMFiles/MMFilesCollectorThread.cpp

```diff
@@ -732,10 +730,10 @@ void MMFilesCollectorThread::processCollectionMarker(
 /// @brief process all operations for a single collection
 int MMFilesCollectorThread::processCollectionOperations(MMFilesCollectorCache* cache) {
   arangodb::DatabaseGuard dbGuard(cache->databaseId);
-  TRI_vocbase_t* vocbase = dbGuard.database();
-  TRI_ASSERT(vocbase != nullptr);
-
-  arangodb::CollectionGuard collectionGuard(vocbase, cache->collectionId, true);
+  auto& vocbase = dbGuard.database();
+  arangodb::CollectionGuard collectionGuard(
```

Comment:
Could we delete the copy constructor for the TRI_vocbase_t ?

Reply:
there is no copy constructor in TRI_vocbase_t, it's non-copyable because of locks

Reply:
@graetzer that is impossible to copy because of the `const` members, .e.g `id`. But I agree, explicit is better than implicit :-)